lib/ansible/galaxy/token.py

```diff
@@ -87,9 +199,31 @@ def get(self):
 
         data = json.loads(to_text(resp.read(), errors='surrogate_or_strict'))
 
-        # - extract 'access_token'
+        if self.jwks is None and self._jwks_url is not None:
+            jwks_resp = open_url(
+                to_native(self._jwks_url),
+                data=payload,
+                validate_certs=self.validate_certs,
+                method='GET',
+                http_agent=user_agent()
+            )
+            self.jwks = json.loads(to_text(jwks_resp.read(), errors='surrogate_or_strict')).get('keys', [])
+
+        # - extract token details
         self._token = data.get('access_token')
 
+        if self.jwks is not None and HAS_CRYPTO:
+            try:
+                token_data = parse_bearer_token(data['id_token'], jwks=self.jwks)
+            except InvalidSignature as e:
+                # something went wrong parsing or validating the JWT, this is unexpected
+                # might need to support other algorithms?
+                # this isn't fatal to keep backwards compat, just won't get to use proactive token refreshing
+                orig_error = f': {e}' if e.args else ''
+                display.warning(f'Failed to verify JWS{orig_error}. The token will not be refreshed during long-running operations.')
+            else:
+                self._expires_at = token_data.get('exp')
```

Comment:
Seems like this block could be moved to a helper function.

Reply:
The current code feels readable to me and since nothing is repeated I don't think a helpful function seems useful here.